modules/solid_mechanics/src/materials/ConstitutiveModel.C

```diff
@@ -93,13 +95,19 @@ ConstitutiveModel::applyThermalStrain(unsigned qp,
                                       SymmTensor & strain_increment,
                                       SymmTensor & d_strain_dT)
 {
-  if (_has_temp && _t_step != 0)
+  if (_t_step >= 1)
+    _zeroth_step = false;
+
+  if (_t_step >= 2)
+    _first_step = false;
+
```

Comment:
I could do a restartable data unsigned int instead of a bool?  Also keep in mind this logic is in solid mechanics SolidModel and ConstitutiveModel, so the awkwardness will go away with solid mechanics; not that solid mechanics is a good excuse to avoid 'doing things right'